pandas/core/generic.py

```diff
@@ -2121,7 +2120,7 @@ def _repr_latex_(self):
         Returns a LaTeX representation for a particular object.
         Mainly for use with nbconvert (jupyter notebook conversion to pdf).
         """
-        if config.get_option("display.latex.repr"):
+        if config.get_option("styler.render.repr") == "latex":
```

Comment:
styler uses different pandas options namespace than dataframe rendering. if a generic call to dataframe.to_latex is made it will now pickup the stylers options (where they exist)  as part of the styler code exectuting.

in the above "display.latex.repr" was no longer being reached.

this can also be overwritten in the method call to divert back to the original options namespace but it seems strange to do that when the documentation is pointing towards styler

Reply:
I'm not sure I'm following. If a user is using 1.4 and sets `display.latex.repr` to True, they would get latex in a notebook. However with this PR if they were to upgrade to 1.5 they would no longer get latex, is that right? They would instead need to change their code to modify `styler.render.repr`?

Reply:
So according to #39911 `display.latex.repr` was broken and still is. But I offered a fix via the `Styler` implementation. The current status is that:
![Screenshot 2022-08-11 at 18 32 13](https://user-images.githubusercontent.com/24256554/184183672-3c6cfe7f-8892-430f-afd0-91c599a68203.png)

Reply:
Isn't this only supposed to have an impact when using nbconvert?

Reply:
https://github.com/pandas-dev/pandas/commit/d20767925fc866f1a51c43424f7b70a023a2240e

Reply:
For full disclosure on this point, JupyterNotebook based on IPython will output data based on `_repr_*_` methods.

For the following class the output is `html` becuase the html method is checked first. 

```python
class A:
    def __init__(self):
        pass
    
    def _repr_markdown_(self):
        return "md"
    
    def _repr_html_(self):
        return "html"
    
    def _repr_latex_(self):
        return "latex"
``` 

If the output for the `_repr_html_` method is `None` it will check the next in priority, which in this case is `markdown`. If the output of `_repr_markdown_` is also `None` it will check `latex`. I.e. the below class would have a JupyterNotebook cell output of `latex`.

```python
class A:
    def __init__(self):
        pass
    
    def _repr_markdown_(self):
        return None
    
    def _repr_html_(self):
        return None
    
    def _repr_latex_(self):
        return "latex"
``` 

The `Styler` object in pandas is set to generate and output only one type (currently either `html` or `latex`), based on the global option `styler.render.repr`. The signature of the Styler class repr methods looks like this:

```python
    def _repr_html_(self) -> str | None:
        if get_option("styler.render.repr") == "html":
            return self.to_html()
        return None

    def _repr_latex_(self) -> str | None:
        if get_option("styler.render.repr") == "latex":
            return self.to_latex()
        return None
```

This means that if one sets the `styler.render.repr` method to `"html"` then both the user *and* nbconvert will see HTML, whilst if "latex" is chosen both the user *and* nbconvert see LaTeX output.

The signature of the `DataFrame` repr methods is different.
If one wants to output any latex at all in Jupyter (which nbconvert will detect) the `display.latex.repr` must be `True`:

```python
    def _repr_latex_(self):
        if config.get_option("display.latex.repr"):
            return self.to_latex()
        else:
            return None
```

If one wants to no longer see HTML in Jupyter, but see the LaTeX as a user, then `display.notebook_repr_html` must also be set to `False` to ensure the `_repr_html_` returns `None`:

```python
    def _repr_html_(self) -> str | None:
        if self._info_repr():
            ...
            return f"<pre>{val}</pre>"

        if get_option("display.notebook_repr_html"):
            ...
            return fmt.DataFrameRenderer(formatter).to_html(notebook=True)
        else:
            return None
```

Personally I think if you want to output to LaTex via nbconvert it is useful to run the notebook as a user and see the LaTeX output anyway so the Styler single option change is better in my opinion but it is exceptional marginal.

Reply:
Thanks for the detailed response here, but I don't understand what the overall implication is. In particular,

> Personally I think if you want to output to LaTex via nbconvert it is useful to run the notebook as a user and see the LaTeX output anyway so the Styler single option change is better in my opinion but it is exceptional marginal.

What does "run the notebook as a user" mean?

Reply:
If you are user executing or running the notebook the cells will execute code on request and render in the browser.
If you are using nbconvert this is a command line auto process that will create a document by executing all cells in sequence in a hidden environment.

If my intention was to create a latex document using nbconvert, I would prefer to run and execute all the cells in the notebook browser to visualise and check the results before using nbconvert to create the document. This can be done by setting `styler.render.repr` to "latex".

To visualise the latex code in the notebook in the old way one needed to do both:
display.latex.repr = True  # turn latex output on
display.notebook_repr_html = False  # turn html output off
Whilst to generate latex using nbconvert it is sufficient to do only display.latex.repr = True.

I think all possibilities are available in each approach its just very slightly differently configured. And I don't think there was any genuine thought into the old structure either. It just naturally evolved as methods were added to the library I expect.

Reply:
> If my intention was to create a latex document using nbconvert, I would prefer to run and execute all the cells in the notebook browser to visualise and check the results before using nbconvert to create the document. This can be done by setting styler.render.repr to "latex".

But with this PR there is no way to generate latex with using nbconvert without manually opening the notebook in a browser?

Reply:
If you set the pandas option styler.render.repr to "latex" it will generate latex using nbconvert or when manually opening the notebook. it does both.

Reply:
Okay - thanks.